clicommand/global.go

```
@@ -150,19 +150,26 @@ func HandleGlobalFlags(l logger.Logger, cfg interface{}) func() {
 	return HandleProfileFlag(l, cfg)
 }
 
-func UnsetConfigFromEnvironment(c *cli.Context) {
+func UnsetConfigFromEnvironment(c *cli.Context) error {
 	flags := append(c.App.Flags, c.Command.Flags...)
 	for _, fl := range flags {
-		// use golang reflection to find EnvVar values on flags
+		// use golang reflection to find EnvVars values on flags
 		r := reflect.ValueOf(fl)
-		f := reflect.Indirect(r).FieldByName(`EnvVar`)
-		// split comma delimited env
-		if envVars := f.String(); envVars != `` {
-			for _, env := range strings.Split(envVars, ",") {
-				os.Unsetenv(env)
-			}
+		f := reflect.Indirect(r).FieldByName(`EnvVars`)
+		if !f.IsValid() {
+			return errors.New("EnvVars field not found on flag")
+		}
+		if f.Kind() != reflect.Slice {
+			return errors.New("EnvVars not a slice")
+		}
```

fail loudly if `ufave/cli/v3` decides to change/rename things again.